cosmos/providers/dbt/core/operators/virtualenv.py

```diff
@@ -0,0 +1,146 @@
+from __future__ import annotations
+
+import logging
+from pathlib import Path
+from tempfile import TemporaryDirectory
+
+from airflow.compat.functools import cached_property
+from airflow.utils.context import Context
```

Comment:
Might be nice to move this import under `TYPE_CHECKING`. One fewer import at runtime. WDYT?